test/plugin/test_out_google_cloud_grpc.rb

```diff
@@ -0,0 +1,955 @@
+        labels: { "#{CONTAINER_SERVICE_NAME}/stream" => 'stderr' }
+      ) { |_, oldval, newval| oldval.merge(newval) }
+      verify_grpc_log_entries(1, expected_params) do |entry|
+        assert_equal CONTAINER_SECONDS_EPOCH, \
+                     entry['metadata']['timestamp']['seconds'], entry
+        assert_equal CONTAINER_NANOS, \
+                     entry['metadata']['timestamp']['nanos'], entry
+        assert_equal 'ERROR', entry['metadata']['severity'], entry
+      end
+    end
+  end
+
+  def test_struct_container_log_metadata_from_plugin_grpc
+    setup_gce_metadata_stubs
+    setup_container_metadata_stubs
+    setup_grpc_logging_stubs do
+      d = create_grpc_driver(USE_GRPC_CONFIG, CONTAINER_TAG)
+      d.emit(container_log_entry_with_metadata('{"msg": "test log entry 0", ' \
+                                               '"tag2": "test", "data": ' \
+                                               '5000, "severity": "WARNING"}'))
+      d.run
+      verify_grpc_log_entries(1, CONTAINER_FROM_METADATA_PARAMS,
+                              'structPayload') do |entry|
+        fields = entry['structPayload']['fields']
+        assert_equal 3, fields.size, entry
+        assert_equal 'test log entry 0', fields['msg']['stringValue'], entry
+        assert_equal 'test', fields['tag2']['stringValue'], entry
+        assert_equal 5000, fields['data']['numberValue'], entry
+        assert_equal CONTAINER_SECONDS_EPOCH, \
+                     entry['metadata']['timestamp']['seconds'], entry
+        assert_equal CONTAINER_NANOS, \
+                     entry['metadata']['timestamp']['nanos'], entry
+        assert_equal 'WARNING', entry['metadata']['severity'], entry
+      end
+    end
+  end
+
+  def test_struct_container_log_metadata_from_tag_grpc
+    setup_gce_metadata_stubs
+    setup_container_metadata_stubs
+    setup_grpc_logging_stubs do
+      d = create_grpc_driver(USE_GRPC_CONFIG, CONTAINER_TAG)
+      d.emit(container_log_entry('{"msg": "test log entry 0", ' \
+                                 '"tag2": "test", "data": 5000, ' \
+                                 '"severity": "W"}'))
+      d.run
+      verify_grpc_log_entries(1, CONTAINER_FROM_TAG_PARAMS,
+                              'structPayload') do |entry|
+        fields = entry['structPayload']['fields']
+        assert_equal 3, fields.size, entry
+        assert_equal 'test log entry 0', fields['msg']['stringValue'], entry
+        assert_equal 'test', fields['tag2']['stringValue'], entry
+        assert_equal 5000, fields['data']['numberValue'], entry
+        assert_equal CONTAINER_SECONDS_EPOCH, \
+                     entry['metadata']['timestamp']['seconds'], entry
+        assert_equal CONTAINER_NANOS, \
+                     entry['metadata']['timestamp']['nanos'], entry
+        assert_equal 'WARNING', entry['metadata']['severity'], entry
+      end
+    end
+  end
+
+  def test_one_cloudfunctions_log_grpc
```

Comment:
Just refactored this. The list is actually just the number of logs we wanna test against. So I combined the list and added 1s to them to represent a single log.